cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringAgentService.java

```diff
@@ -102,31 +103,30 @@
   private final TransactionRunner transactionRunner;
   private final Map<String, String> lastMessageIds;
   private final ProgramStateWriter programStateWriter;
-  private final MessageFetcher messageFetcher;
   private final ProgramRunRecordFetcher runRecordFetcher;
   private final RemoteAuthenticator remoteAuthenticator;
-  private final String programUpdateTopic;
-  private final int programUpdateFetchSize;
-  private String lastProgramUpdateMessageId;
+  private final RetryStrategy retryStrategy;
 
   @Inject
   TetheringAgentService(CConfiguration cConf, TransactionRunner transactionRunner, TetheringStore store,
                         ProgramStateWriter programStateWriter, MessagingService messagingService,
                         ProgramRunRecordFetcher programRunRecordFetcher,
+                        MetricsCollectionService metricsCollectionService,
                         @Named(REMOTE_TETHERING_AUTHENTICATOR) RemoteAuthenticator remoteAuthenticator) {
-    super(RetryStrategies.fromConfiguration(cConf, "tethering.agent."));
+    super("tethering.agent", cConf, cConf.get(Constants.AppFabric.PROGRAM_STATUS_RECORD_EVENT_TOPIC),
+          cConf.getInt(Constants.AppFabric.STATUS_EVENT_FETCH_SIZE),
+          cConf.getLong(Constants.AppFabric.STATUS_EVENT_POLL_DELAY_MILLIS), messagingService,
+          metricsCollectionService, transactionRunner);
+    this.retryStrategy = RetryStrategies.fromConfiguration(cConf, "tethering.agent.");
     this.connectionInterval = TimeUnit.SECONDS.toMillis(cConf.getLong(Constants.Tethering.CONNECTION_INTERVAL));
     this.cConf = cConf;
     this.transactionRunner = transactionRunner;
     this.store = store;
     this.instanceName = cConf.get(Constants.INSTANCE_NAME);
     this.lastMessageIds = new HashMap<>();
```

Comment:
think this can be removed

Reply:
this is used when connecting to tethering server (so the tethering server knows what was the last control message that was sent over to this agent)
https://github.com/cdapio/cdap/blob/0e20a3f77047ea4586a377edc913ded59e328ea8/cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringServerHandler.java#L119-L138

Reply:
added a comment to describe the use case and renamed the variable to avoid confusion